parser-typechecker/src/Unison/Codebase/FileCodebase.hs

```diff
@@ -588,15 +588,21 @@ termReferencesByPrefix root = loadReferencesByPrefix (termsDir root)
 typeReferencesByPrefix root = loadReferencesByPrefix (typesDir root)
 
 -- returns all the derived terms and derived constructors
+-- that have `sh` as a prefix
 termReferentsByPrefix :: MonadIO m
   => (CodebasePath -> Reference.Id -> m (Maybe (DD.Decl v a)))
   -> CodebasePath
   -> ShortHash
   -> m (Set Referent.Id)
-termReferentsByPrefix getDecl root sh = do
+termReferentsByPrefix _ root sh@SH.Builtin{} =
+  Set.map Referent.Ref' <$> termReferencesByPrefix root sh
+  -- builtin types don't provide any referents we could match against,
+  -- only decl types do.  
+termReferentsByPrefix getDecl root sh@SH.ShortHash{} = do
   terms <- termReferencesByPrefix root sh
   ctors <- do
-    types <- typeReferencesByPrefix root sh
+    -- clear out any CID from the SH, so we can use it to find a type decl
+    types <- typeReferencesByPrefix root sh { SH.cid = Nothing }
     foldMapM collectCtors types
   pure (Set.map Referent.Ref' terms <> ctors)
```

Comment:
This and the next chunk is the fix for #1399.  It was causing all constructors to show up in a search for a type decl, but they shouldn't.  They should only show up if they match a constructor id in the search query shorthash.